File: sol/solution/0001-0100/0019/Solution.ex
# Definition for singly-linked list.
#
# defmodule ListNode do
#   @type t :: %__MODULE__{
#           val: integer,
#           next: ListNode.t() | nil
#         }
#   defstruct val: 0, next: nil
# end
defmodule Solution do
  @spec remove_nth_from_end(head :: ListNode.t | nil, n :: integer) :: ListNode.t | nil
  def remove_nth_from_end(head, n), do: remove_nth(head, head, 0, n)
  defp remove_nth(nil, slow, _i, _n), do: slow.next
  defp remove_nth(fast, slow, i, n) when i >= n do
    %ListNode{val: slow.val, next: remove_nth(fast.next, slow.next, i + 1, n)}
  end
  defp remove_nth(fast, slow, i, n), do: remove_nth(fast.next, slow, i + 1, n)
end

File: sol/solution/0001-0100/0019/Solution.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
var removeNthFromEnd = function(head, n) {
    var dummy = new ListNode(0, head);
    var fast = dummy;
    var slow = dummy;
    for (var i = 0; i < n; i++) {
        fast = fast.next;
    }
    while (fast.next !== null) {
        fast = fast.next;
        slow = slow.next;
    }
    slow.next = slow.next.next;
    return dummy.next;
};